main.tf

```diff
@@ -35,14 +35,15 @@ resource "aws_instance" "this" {
   ebs_optimized = var.ebs_optimized
 
   dynamic "capacity_reservation_specification" {
```

Comment:
this was the issue and it now matches https://github.com/terraform-aws-modules/terraform-aws-ec2-instance/blob/dbf16a26e3077ac1b9ebc7f674f81bd710c438fa/main.tf#L184-L197